### ISSUE TITLE
Would this work with Regex?

### ISSUE THREAD

**Issue by FM1337** [JoshuaDoes/gofuckyourself#1]
Let's say I have a list like this https://raw.githubusercontent.com/kwsch/PKHeX/master/PKHeX.Core/Resources/text/locale/badwords.txt

Would this work with your swear filter library?

**Comment by FM1337** [comment]
nvm this was a stupid question

**Comment by JoshuaDoes** [comment]
I'm afraid it won't work as it currently stands, however that's not to say it can't work o:

You just gave me a great idea: Adding pure regular expression support, as simply creating a list of words in an array will still count as a valid list of regular expressions in an array to the regex parser.

I'm going to re-open this for the time being to make sure I keep my attention to it and can get around to this sometime soon

**Comment by boosh** [comment]
Implemented in https://github.com/boosh/swearfilter

**Comment by JoshuaDoes** [comment]
> Implemented in https://github.com/boosh/swearfilter

It's MIT but I still wanna ask, mind if I cherry pick your commits? I'll edit them before pushing to remove repo-breaking changes and follow it with a credits list commit for you and a previous fork that I gave the same treatment to, as well as keep full authorship for your commits (as you can see in the commit history for contributors).

I'm also going to consider renaming the repo to be a little more professional, as it seems to be a primary reason for people to fork the repo (so I imagine cloners are renaming their imports too). I don't think I need to explain the joke behind the current name, but it might help if I make it match the package name to begin with.

**Comment by boosh** [comment]
Yeah no worries. The issue with the repo name wasn't the name itself, it's that because the actual golang package name is different to the GH repo name it was tripping up using it in a go.mod file. It's just easier to rename the repo to match the golang package name.
